### PR TITLE
add group anagram algorithm

### PR DESCRIPTION
**Pull request by Kishore-abhimanyu**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/group_anagram.py
Comment (Kishore-abhimanyu) [reply strings/group_anagram.py:4]: I have added the doctest

**Comment by Kishore-abhimanyu** [comment]
I have resolved this issue

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please can you add more comments and docstrings so the reader understands what is happening at all times

**Comment by Kishore-abhimanyu** [comment]
> Please can you add more comments and docstrings so the reader understands what is happening at all times

Comment are added

File: strings/group_anagram.py
Comment (Kishore-abhimanyu) [reply strings/group_anagram.py:4]: doctest is also added with the name group_anagram.txt

**Comment by Kishore-abhimanyu** [comment]
Can you check now

File: strings/group_anagram.py
Comment (cclauss) [inline strings/group_anagram.py:39]:
```suggestion
        _set = sorted(set(res))
```

File: strings/group_anagram.py
Comment (cclauss) [inline strings/group_anagram.py:23]:
```suggestion
    _keys = list(char_map)
```

File: strings/group_anagram.py
Comment (cclauss) [reply strings/group_anagram.py:4]: This function does NOT have a doctes.

File: strings/group_anagram.py
Comment (cclauss) [inline strings/group_anagram.py:11]:
```suggestion
        char_map[each] = {"is_visited": False}
```

File: strings/group_anagram.py
Comment (cclauss) [inline strings/group_anagram.py:32]:
```suggestion
                and not char_map[_keys[itr2]]["is_visited"]
```

File: strings/group_anagram.py
Comment (cclauss) [inline strings/group_anagram.py:37]:
```suggestion
                char_map[_keys[itr2]]["is_visited"] = True
```

File: strings/group_anagram.py
Comment (cclauss) [inline strings/group_anagram.py:48]: Put the test in the function so that if someone copies the function into their code, the test comes with it.

**Comment by Kishore-abhimanyu** [comment]
The changes has been updated. Could you check the changes

File: strings/group_anagram.py
Comment (cclauss) [inline strings/group_anagram.py:13]: Why is this an improvement?

Comment (Kishore-abhimanyu) [reply strings/group_anagram.py:13]: If char_map[each] is responsible for both tacking the visitors and the count, if it is referenced before assignment, there may be a cause of runtime error

**Comment by cclauss** [comment]
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

strings/group_anagram.py:9: error: Need type annotation for "char_map" (hint: "char_map: Dict[<type>, <type>] = ...")

**Comment by cclauss** [comment]
Please read CONTRIBUTING.md.

Hacktoberfest is strenuous for our team of active volunteer maintainers. That small team now has ___200+ open pull requests___ to review.  We have __high standards for our submissions__ which are described in [`CONTRIBUTING.md`](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md#contributing-guidelines).  Our continuous integration tests help to ensure that our standards are met before we review and merge deserving pull requests.  This is further complicated by the fact that we already have [lots of algorithms](https://github.com/TheAlgorithms/Python/blob/master/DIRECTORY.md) and would like to avoid duplicates.

For these reasons, I will **close all open pull requests failing tests** so that our unpaid team has a smaller mountain to climb.  If you want us to land your pull request, please read [`CONTRIBUTING.md`](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md#contributing-guidelines) _carefully_ and try again.

**Pull request closed without merging**
